packages/example-contracts/contracts/cross-chain-message/CrossChainMessage.sol

```diff
@@ -20,11 +20,18 @@ contract CrossChainMessage is ZetaInteractor, ZetaReceiver, CrossChainMessageErr
     event HelloWorldEvent(string messageData);
     event RevertedHelloWorldEvent(string messageData);
 
-    constructor(address connectorAddress_) ZetaInteractor(connectorAddress_) {}
+    ZetaTokenConsumer private _zetaConsumer;
 
-    function sendHelloWorld(uint256 destinationChainId) external {
+    constructor(address connectorAddress_, address zetaConsumerAddress) ZetaInteractor(connectorAddress_) {
+        _zetaConsumer = ZetaTokenConsumer(zetaConsumerAddress);
+    }
+
+    function sendHelloWorld(uint256 destinationChainId) external payable {
         if (!_isValidChainId(destinationChainId)) revert InvalidDestinationChainId();
 
+        uint256 crossChainGas = 18000000000000000000;
+        _zetaConsumer.getZetaFromEth{value: msg.value}(address(this), crossChainGas);
+
```

Comment:
We need to update `zetaValueAndGas`, otherwise the contract still won't work.
Also, we should find a way in which we test that some gas is being consumed so tests fail if we pass `zetaValueAndGas: 0`.

Reply:
We also need to `approve` the Connector with the amount of Zeta we got, so it can be burnt.